CANTroller2/src/IdiotLight.h

```diff
@@ -0,0 +1,92 @@
+#ifndef IDIOT_LIGHT_H
+#define IDIOT_LIGHT_H
+
+#include "globals.h"
+#include <map>
+#include "TemperatureSensorManager.h"
+
+class IdiotLight {
+public:
+    using StateFunction = std::function<bool()>;
+
+    IdiotLight(StateFunction state_func, uint16_t color, const char* label)
+        : state_func(state_func), color(color), label(label), last_state(state_func()) {}
+
+    bool has_changed() const {
+        return state_func() != last_state;
+    }
+
+    void update() {
+        last_state = state_func();
+    }
+
+    bool get_state() const {
+        return state_func();
+    }
+
+    uint16_t get_color() const {
+        return color;
+    }
+
+    const char* get_label() const {
+        return label;
+    }
+
+private:
+    StateFunction state_func;
+    uint16_t color;
+    const char* label;
+    bool last_state;
+};
+
+enum class IdiotLightName {
+    START, REMOTE_START, IGNITION_SENSE, IGNITION, SYSPOWER, SHUTDOWN_COMPLETE, MODE_SWITCH, SIMULATOR_ENABLED, RADIO_DETECTED, PANIC_STOP, MOTORS_PARKED, CRUISE_ADJUSTING, ENGINE_TEMP_WARN, ENGINE_TEMP_ALARM, WHEEL_TEMP_WARN, WHEEL_TEMP_ALARM };
+
+// we can put all of the more complicated state functions here to determine if the light should be on or not. 
+bool check_engine_temp_warn() {
+    TemperatureSensor* engine_sensor = temperature_sensor_manager.get_sensor(TemperatureSensor::location::ENGINE);
+    return engine_sensor->get_temperature() >= engine_sensor->get_limits().get_warning();
+}
+
+bool check_engine_temp_alarm() {
+    TemperatureSensor* engine_sensor = temperature_sensor_manager.get_sensor(TemperatureSensor::location::ENGINE);
+    return engine_sensor->get_temperature() >= engine_sensor->get_limits().get_alarm();
+}
+
+bool check_wheel_temp_warn() {
+    TemperatureSensor* wheel_sensor = temperature_sensor_manager.get_sensor(TemperatureSensor::location::WHEEL_FL);
+    return wheel_sensor->get_temperature() >= wheel_sensor->get_limits().get_warning();
+}
+
+bool check_wheel_temp_alarm() {
+    TemperatureSensor* wheel_sensor = temperature_sensor_manager.get_sensor(TemperatureSensor::location::WHEEL_FL);
+    return wheel_sensor->get_temperature() >= wheel_sensor->get_limits().get_alarm();
+}
+
+std::map<IdiotLightName, IdiotLight> idiot_lights = {
+    {IdiotLightName::START, IdiotLight([](){return starter;}, GRN, "St")},
+    {IdiotLightName::REMOTE_START, IdiotLight([](){return remote_starting;}, TEAL, "RS")},
+    {IdiotLightName::IGNITION_SENSE, IdiotLight([](){return ignition_sense;}, ORG, "Is")},
+    {IdiotLightName::IGNITION, IdiotLight([](){return ignition;}, YEL, "IG")},
+    {IdiotLightName::SYSPOWER, IdiotLight([](){return syspower;}, GRN, "Pw")},
+    {IdiotLightName::SHUTDOWN_COMPLETE, IdiotLight([](){return shutdown_complete;}, ORG, "Sh")},
+    {IdiotLightName::MODE_SWITCH, IdiotLight([](){return we_just_switched_modes;}, LYEL, "We")},
+    {IdiotLightName::SIMULATOR_ENABLED, IdiotLight([](){return simulator.get_enabled_ptr();}, PNK, "Sm")},
```

Comment:
i think this is wrong, you want the actual value and not the pointer. so just simulator.get_enabled() should work fine (using these function makes interfacing with idiot lights much nicer for Device and friends).